### DISCUSSION TITLE
Deletion of __splink tables causes the developed code to fail on the estiamate_u step

### DISCUSSION THREAD

**Discussion by sthamodh** [moj-analytical-services/splink#2039] [General]
Hi all,

I was able to figure out that splink creates tables by the name "__splink" and writes them to a default place. 
In my case, I use databricks with a 12.2 LTS ML compute and was able to figure out that these table are being written to the dbfs location,

"dbfs:/user/hive/metastore"

I then proceeded to manually delete these tables (as there were 100s of them from previous runs) using the code below:

![deletion_command](https://github.com/moj-analytical-services/splink/assets/156929377/8fe9d0f2-5b7a-486d-982e-f5b796d35989)

Once these tables were deleted, my previously tested solution fails on the estimating u step as shown below:

![for_github](https://github.com/moj-analytical-services/splink/assets/156929377/9104c282-5efe-4dec-b954-6496933962bc)

As highlighted in the red box above, I added the _set_catalog_and_database_if_not_provided function and specified where I would like these intermediate tables to be written.

Error:

![table_does_not_exist](https://github.com/moj-analytical-services/splink/assets/156929377/cb90b01b-a827-4111-9a85-481c7747db1e)

1. How would one make sure that intermediate tables are created in a specific place and it does not write these "__splink" tables to a default space?
2. How would one delete these tables after each run? Is there a method for the linker object that can do this?

**Comment by ADBond** [comment]
Without having looked in much detail, I think the issue you are running into here will be to do with caching. For performance reasons Splink keeps information about tables it has already computed so that if they are required at other steps of the pipeline they can be read from the db rather than re-computed. So I think the trouble is that you delete all the Splink tables, but the linker does not know that these tables no longer exist, so you hit an error when it tries to read from one.

There is a linker method that will handle this for you: `linker.delete_tables_created_by_splink_from_db()`, which will also delete them from the cache. If you ever want to clear the cache without deleting tables from the database you can run `linker.invalidate_cache()`.

Also worth mentioning that if you create a new linker object, this will use a separate cache + a different set of tables, which might be useful if you need to do separate linkings without deleting everything.
